store/src/java/com/zimbra/cs/mailbox/MailItem.java

```diff
@@ -2753,7 +2762,11 @@ MailItem icopy(Folder target, int copyId, String copyUuid) throws IOException, S
         DbMailItem.icopy(this, data, shareIndex);
 
         MailItem copy = constructItem(mMailbox, data);
-        copy.finishCreation(null);
+        if (target.getId() == FolderConstants.ID_FOLDER_TRASH) {
+            copy.finishCreation(null, true);
+        } else {
+            copy.finishCreation(null);
+        }
```

Comment:
Can we simplify it without if-else as well? Is the following code less readable? What do you think?
`copy.finishCreation(null, target.getId() == FolderConstants.ID_FOLDER_TRASH);`

Reply:
This is much readable and also removed unwanted code , updated as per suggestion